NDTensors/src/lib/SmallVectors/src/abstractsmallvector/deque.jl

```diff
@@ -311,7 +311,7 @@ function prepend(vec::AbstractSmallVector, item::AbstractVector)
 end
 
 # Don't @inline, makes it slower.
-function Base.vcat(vec1::AbstractSmallVector, vec2::AbstractVector)
+function Base.vcat(vec1::AbstractSmallVector{<:Number}, vec2::AbstractVector{<:Number})
```

Comment:
This doesn't seem great since we generally will not only have `Number`s inside of a `SmallVector`.

Something we can do is:
```julia
function smallvector_vcat(vec1::AbstractSmallVector, vec2::AbstractVector)
  # Current definition
end

Base.vcat(vec1::AbstractSmallVector{<:Number}, vec2::AbstractVector{<:Number}) = smallvector_vcat(vec1, vec2)

Base.vcat(vec1::AbstractSmallVector, vec2::AbstractVector) = smallvector_vcat(vec1, vec2)
```